src/core/thread/mle.cpp

```diff
@@ -1497,8 +1510,23 @@ ThreadError Mle::HandleDataResponse(const Message &aMessage, const Ip6::MessageI
     SuccessOrExit(error = Tlv::GetTlv(aMessage, Tlv::kLeaderData, sizeof(leaderData), leaderData));
     VerifyOrExit(leaderData.IsValid(), error = kThreadError_Parse);
 
-    diff = leaderData.GetDataVersion() - mNetworkData.GetVersion();
-    VerifyOrExit(diff > 0, ;);
+    if (mRetrieveNewNetworkData)
+    {
+        mRetrieveNewNetworkData = false;
+    }
+    else
+    {
+        diff = leaderData.GetDataVersion() - mNetworkData.GetVersion();
+
+        if (((mDeviceMode & ModeTlv::kModeRxOnWhenIdle) == 0) && (diff != 0))
+        {
+            SetLeaderData(leaderData.GetPartitionId(), leaderData.GetWeighting(), leaderData.GetLeaderRouterId());
```

Comment:
Ah, yes.  I think that makes sense.  We might want to add a check for rx-on-when-idle children to ensure that the partition ID value is as expected.